Generate vendor and modifier enums

Drakulix opened the pull request:
Closes #1 and #2.

Mimics the generation of `format` for `vendor` and `modifier` with slight changes on the later.
Because of several modifier combinations, that cannot be easily generated, and overlapping names for the same underlying modifier, the resulting enum does not use `repr` and contains an `Other(u64)` variant.

Otherwise I tried to model everything after the existing code. The `build.rs` file is now a bit repetitive, but since the modifier generation is always slightly off, de-duplicating the `vendor` and `format` generation seemed not very helpful.
It is still quite readable although a bit verbose.

If you have any suggestions on how to address any of these short-comings, or anything else regarding this PR, just tell me. :)

danielzfranklin commented:
This looks great! I've got a few thoughts

I wonder if things would fit together better if we renamed `DrmFormat` to `DrmFourcc` and replaced it with a`DrmFormat(DrmFourcc, DrmModifier)`. What do you think?

If `DrmVendor:: try_from` returns a `Result<Option<DrmVendor>, ...>` it removes the duplication in the two `::vendor` fns.

danielzfranklin commented:
One more minor thing I noticed: When I generate the bindings on my laptop I'm missing a few of the modifiers you generated. Please update the note in the README about the kernel version the bindings were generated from with your kernel version.

danielzfranklin commented on build.rs:
```diff
@@ -119,16 +160,76 @@ mod generate {
+
+            as_enum.write_all(b"impl DrmModifier {\n")?;
+            as_enum.write_all(b"pub(crate) fn from_u64(n: u64) -> Self {\n")?;
+            as_enum.write_all(b"#[allow(unreachable_patterns)]\n")?;
```
I don't see any warning without this. Why is it necessary?

Drakulix replied:
Because they can overlap as some vendor-specific modifiers might just be generic-modifiers.

E.g. with my kernel version, I am getting:
```
warning: unreachable pattern
   --> src/as_enum.rs:316:13
    |
316 |             consts::DRM_FOURCC_SAMSUNG_16_16_TILE => Self::Samsung_16_16_tile,
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unreachable_patterns)]` on by default

warning: 1 warning emitted
```
Because `DRM_FOURCC_SAMSUNG_16_16_TILE` has the same value as `DRM_FOURCC_GENERIC_16_16_TILE`.

That means the match statement will contain the same value two times, which makes the second match unreachable. Currently sorting defines, which match arm is first and will always return, but because of the custom `PartialEq` implementation, I figured that it does not really matter.

Drakulix commented:
> I wonder if things would fit together better if we renamed `DrmFormat` to `DrmFourcc` and replaced it with a`DrmFormat(DrmFourcc, DrmModifier)`. What do you think?

Good idea.

> If `DrmVendor:: try_from` returns a `Result<Option<DrmVendor>, ...>` it removes the duplication in the two `::vendor` fns.

Yeah, but that would mean implementing `TryFrom` for `Option<DrmVendor>`, which is impossible. Or am I missing something here?

> One more minor thing I noticed: When I generate the bindings on my laptop I'm missing a few of the modifiers you generated. Please update the note in the README about the kernel version the bindings were generated from with your kernel version.

Sure.

danielzfranklin commented:
Looks great! I'll add your name to the readme and publish tomorrow.
